// react-magic-motion/dist/index.mjs
var __create = Object.create;
var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
var __commonJS = (cb, mod) => function __require() {
  return mod || (0, cb[__getOwnPropNames(cb)[0]])((mod = { exports: {} }).exports, mod), mod.exports;
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));

// ../../node_modules/.pnpm/react-is@18.2.0/node_modules/react-is/cjs/react-is.production.min.js
var require_react_is_production_min = __commonJS({
  "../../node_modules/.pnpm/react-is@18.2.0/node_modules/react-is/cjs/react-is.production.min.js"(exports) {
    "use strict";
    var b = Symbol.for("react.element");
    var c = Symbol.for("react.portal");
    var d = Symbol.for("react.fragment");
    var e = Symbol.for("react.strict_mode");
    var f = Symbol.for("react.profiler");
    var g = Symbol.for("react.provider");
    var h = Symbol.for("react.context");
    var k = Symbol.for("react.server_context");
    var l = Symbol.for("react.forward_ref");
    var m4 = Symbol.for("react.suspense");
    var n = Symbol.for("react.suspense_list");
    var p = Symbol.for("react.memo");
    var q = Symbol.for("react.lazy");
    var t = Symbol.for("react.offscreen");
    var u;
    u = Symbol.for("react.module.reference");
    function v(a) {
      if ("object" === typeof a && null !== a) {
        var r = a.$$typeof;
        switch (r) {
          case b:
            switch (a = a.type, a) {
              case d:
              case f:
              case e:
              case m4:
              case n:
                return a;
              default:
                switch (a = a && a.$$typeof, a) {
                  case k:
                  case h:
                  case l:
                  case q:
                  case p:
                  case g:
                    return a;
                  default:
                    return r;
                }
            }
          case c:
            return r;
        }
      }
    }
    exports.ContextConsumer = h;
    exports.ContextProvider = g;
    exports.Element = b;
    exports.ForwardRef = l;
    exports.Fragment = d;
    exports.Lazy = q;
    exports.Memo = p;
    exports.Portal = c;
    exports.Profiler = f;
    exports.StrictMode = e;
    exports.Suspense = m4;
    exports.SuspenseList = n;
    exports.isAsyncMode = function() {
      return false;
    };
    exports.isConcurrentMode = function() {
      return false;
    };
    exports.isContextConsumer = function(a) {
      return v(a) === h;
    };
    exports.isContextProvider = function(a) {
      return v(a) === g;
    };
    exports.isElement = function(a) {
      return "object" === typeof a && null !== a && a.$$typeof === b;
    };
    exports.isForwardRef = function(a) {
      return v(a) === l;
    };
    exports.isFragment = function(a) {
      return v(a) === d;
    };
    exports.isLazy = function(a) {
      return v(a) === q;
    };
    exports.isMemo = function(a) {
      return v(a) === p;
    };
    exports.isPortal = function(a) {
      return v(a) === c;
    };
    exports.isProfiler = function(a) {
      return v(a) === f;
    };
    exports.isStrictMode = function(a) {
      return v(a) === e;
    };
    exports.isSuspense = function(a) {
      return v(a) === m4;
    };
    exports.isSuspenseList = function(a) {
      return v(a) === n;
    };
    exports.isValidElementType = function(a) {
      return "string" === typeof a || "function" === typeof a || a === d || a === f || a === e || a === m4 || a === n || a === t || "object" === typeof a && null !== a && (a.$$typeof === q || a.$$typeof === p || a.$$typeof === g || a.$$typeof === h || a.$$typeof === l || a.$$typeof === u || void 0 !== a.getModuleId) ? true : false;
    };
    exports.typeOf = v;
  }
});

// ../../node_modules/.pnpm/react-is@18.2.0/node_modules/react-is/cjs/react-is.development.js
var require_react_is_development = __commonJS({
  "../../node_modules/.pnpm/react-is@18.2.0/node_modules/react-is/cjs/react-is.development.js"(exports) {
    "use strict";
    if (process.env.NODE_ENV !== "production") {
      (function() {
        "use strict";
        var REACT_ELEMENT_TYPE = Symbol.for("react.element");
        var REACT_PORTAL_TYPE = Symbol.for("react.portal");
        var REACT_FRAGMENT_TYPE = Symbol.for("react.fragment");
        var REACT_STRICT_MODE_TYPE = Symbol.for("react.strict_mode");
        var REACT_PROFILER_TYPE = Symbol.for("react.profiler");
        var REACT_PROVIDER_TYPE = Symbol.for("react.provider");
        var REACT_CONTEXT_TYPE = Symbol.for("react.context");
        var REACT_SERVER_CONTEXT_TYPE = Symbol.for("react.server_context");
        var REACT_FORWARD_REF_TYPE = Symbol.for("react.forward_ref");
        var REACT_SUSPENSE_TYPE = Symbol.for("react.suspense");
        var REACT_SUSPENSE_LIST_TYPE = Symbol.for("react.suspense_list");
        var REACT_MEMO_TYPE = Symbol.for("react.memo");
        var REACT_LAZY_TYPE = Symbol.for("react.lazy");
        var REACT_OFFSCREEN_TYPE = Symbol.for("react.offscreen");
        var enableScopeAPI = false;
        var enableCacheElement = false;
        var enableTransitionTracing = false;
        var enableLegacyHidden = false;
        var enableDebugTracing = false;
        var REACT_MODULE_REFERENCE;
        {
          REACT_MODULE_REFERENCE = Symbol.for("react.module.reference");
        }
        function isValidElementType(type) {
          if (typeof type === "string" || typeof type === "function") {
            return true;
          }
          if (type === REACT_FRAGMENT_TYPE || type === REACT_PROFILER_TYPE || enableDebugTracing || type === REACT_STRICT_MODE_TYPE || type === REACT_SUSPENSE_TYPE || type === REACT_SUSPENSE_LIST_TYPE || enableLegacyHidden || type === REACT_OFFSCREEN_TYPE || enableScopeAPI || enableCacheElement || enableTransitionTracing) {
            return true;
          }
          if (typeof type === "object" && type !== null) {
            if (type.$$typeof === REACT_LAZY_TYPE || type.$$typeof === REACT_MEMO_TYPE || type.$$typeof === REACT_PROVIDER_TYPE || type.$$typeof === REACT_CONTEXT_TYPE || type.$$typeof === REACT_FORWARD_REF_TYPE || // This needs to include all possible module reference object
            // types supported by any Flight configuration anywhere since
            // we don't know which Flight build this will end up being used
            // with.
            type.$$typeof === REACT_MODULE_REFERENCE || type.getModuleId !== void 0) {
              return true;
            }
          }
          return false;
        }
        function typeOf(object) {
          if (typeof object === "object" && object !== null) {
            var $$typeof = object.$$typeof;
            switch ($$typeof) {
              case REACT_ELEMENT_TYPE:
                var type = object.type;
                switch (type) {
                  case REACT_FRAGMENT_TYPE:
                  case REACT_PROFILER_TYPE:
                  case REACT_STRICT_MODE_TYPE:
                  case REACT_SUSPENSE_TYPE:
                  case REACT_SUSPENSE_LIST_TYPE:
                    return type;
                  default:
                    var $$typeofType = type && type.$$typeof;
                    switch ($$typeofType) {
                      case REACT_SERVER_CONTEXT_TYPE:
                      case REACT_CONTEXT_TYPE:
                      case REACT_FORWARD_REF_TYPE:
                      case REACT_LAZY_TYPE:
                      case REACT_MEMO_TYPE:
                      case REACT_PROVIDER_TYPE:
                        return $$typeofType;
                      default:
                        return $$typeof;
                    }
                }
              case REACT_PORTAL_TYPE:
                return $$typeof;
            }
          }
          return void 0;
        }
        var ContextConsumer = REACT_CONTEXT_TYPE;
        var ContextProvider = REACT_PROVIDER_TYPE;
        var Element = REACT_ELEMENT_TYPE;
        var ForwardRef = REACT_FORWARD_REF_TYPE;
        var Fragment3 = REACT_FRAGMENT_TYPE;
        var Lazy = REACT_LAZY_TYPE;
        var Memo = REACT_MEMO_TYPE;
        var Portal = REACT_PORTAL_TYPE;
        var Profiler = REACT_PROFILER_TYPE;
        var StrictMode = REACT_STRICT_MODE_TYPE;
        var Suspense = REACT_SUSPENSE_TYPE;
        var SuspenseList = REACT_SUSPENSE_LIST_TYPE;
        var hasWarnedAboutDeprecatedIsAsyncMode = false;
        var hasWarnedAboutDeprecatedIsConcurrentMode = false;
        function isAsyncMode(object) {
          {
            if (!hasWarnedAboutDeprecatedIsAsyncMode) {
              hasWarnedAboutDeprecatedIsAsyncMode = true;
              console["warn"]("The ReactIs.isAsyncMode() alias has been deprecated, and will be removed in React 18+.");
            }
          }
          return false;
        }
        function isConcurrentMode(object) {
          {
            if (!hasWarnedAboutDeprecatedIsConcurrentMode) {
              hasWarnedAboutDeprecatedIsConcurrentMode = true;
              console["warn"]("The ReactIs.isConcurrentMode() alias has been deprecated, and will be removed in React 18+.");
            }
          }
          return false;
        }
        function isContextConsumer2(object) {
          return typeOf(object) === REACT_CONTEXT_TYPE;
        }
        function isContextProvider2(object) {
          return typeOf(object) === REACT_PROVIDER_TYPE;
        }
        function isElement(object) {
          return typeof object === "object" && object !== null && object.$$typeof === REACT_ELEMENT_TYPE;
        }
        function isForwardRef2(object) {
          return typeOf(object) === REACT_FORWARD_REF_TYPE;
        }
        function isFragment(object) {
          return typeOf(object) === REACT_FRAGMENT_TYPE;
        }
        function isLazy(object) {
          return typeOf(object) === REACT_LAZY_TYPE;
        }
        function isMemo(object) {
          return typeOf(object) === REACT_MEMO_TYPE;
        }
        function isPortal2(object) {
          return typeOf(object) === REACT_PORTAL_TYPE;
        }
        function isProfiler2(object) {
          return typeOf(object) === REACT_PROFILER_TYPE;
        }
        function isStrictMode(object) {
          return typeOf(object) === REACT_STRICT_MODE_TYPE;
        }
        function isSuspense(object) {
          return typeOf(object) === REACT_SUSPENSE_TYPE;
        }
        function isSuspenseList(object) {
          return typeOf(object) === REACT_SUSPENSE_LIST_TYPE;
        }
        exports.ContextConsumer = ContextConsumer;
        exports.ContextProvider = ContextProvider;
        exports.Element = Element;
        exports.ForwardRef = ForwardRef;
        exports.Fragment = Fragment3;
        exports.Lazy = Lazy;
        exports.Memo = Memo;
        exports.Portal = Portal;
        exports.Profiler = Profiler;
        exports.StrictMode = StrictMode;
        exports.Suspense = Suspense;
        exports.SuspenseList = SuspenseList;
        exports.isAsyncMode = isAsyncMode;
        exports.isConcurrentMode = isConcurrentMode;
        exports.isContextConsumer = isContextConsumer2;
        exports.isContextProvider = isContextProvider2;
        exports.isElement = isElement;
        exports.isForwardRef = isForwardRef2;
        exports.isFragment = isFragment;
        exports.isLazy = isLazy;
        exports.isMemo = isMemo;
        exports.isPortal = isPortal2;
        exports.isProfiler = isProfiler2;
        exports.isStrictMode = isStrictMode;
        exports.isSuspense = isSuspense;
        exports.isSuspenseList = isSuspenseList;
        exports.isValidElementType = isValidElementType;
        exports.typeOf = typeOf;
      })();
    }
  }
});

// ../../node_modules/.pnpm/react-is@18.2.0/node_modules/react-is/index.js
var require_react_is = __commonJS({
  "../../node_modules/.pnpm/react-is@18.2.0/node_modules/react-is/index.js"(exports, module) {
    "use strict";
    if (process.env.NODE_ENV === "production") {
      module.exports = require_react_is_production_min();
    } else {
      module.exports = require_react_is_development();
    }
  }
});

// magic-card/index.tsx
import {
  AnimatePresence,
  LazyMotion,
  domMax,
  m as m2
} from "framer-motion";
import { Children as Children2, createElement as createElement2, isValidElement as isValidElement2, useRef } from "react";

// hooks.tsx
import { useEffect, useState } from "react";

// utils/logging.ts
var KEYEXCLUDEMESSAGE = "\u{1F446} Above element is excluded from being animated";
var FORBIDDENELEMENTMESSAGE = (name) => `Forbidden element encountered: ${name} 
Stopping traversal!`;
var FUNCTIONCOMPONENTMESSAGE = (name) => `Function component encountered: ${name}`;
function logSuccess(message) {
  console.log("%cSuccess: %s", "color: green; font-weight: bold;", message);
}
function logWarning(message) {
  console.log("%cWarning: %s", "color: darkorange; font-weight: bold;", message);
}

// hooks.tsx
var PREFERSREDUCEDMOTION = "(prefers-reduced-motion: no-preference)";
function usePlaceholderBoxSize(isCardExpanded, rootNode) {
  const [placeholderBoxDimensions, setPlaceholderBoxDimensions] = useState({
    width: 0,
    height: 0
  });
  useEffect(() => {
    if (!isCardExpanded && rootNode.current) {
      const resizeObserver = new ResizeObserver((entries) => {
        const dimensions = {
          width: entries[0].borderBoxSize[0].inlineSize,
          height: entries[0].borderBoxSize[0].blockSize
        };
        setPlaceholderBoxDimensions({
          width: dimensions.width,
          height: dimensions.height
        });
      });
      resizeObserver.observe(rootNode.current);
      return () => {
        resizeObserver.disconnect();
      };
    }
  }, [isCardExpanded, rootNode]);
  return {
    placeholderBoxHeight: placeholderBoxDimensions.height,
    placeholderBoxWidth: placeholderBoxDimensions.width
  };
}
function usePrefersReducedMotion() {
  const [prefersReducedMotion, setPrefersReducedMotion] = useState(false);
  useEffect(() => {
    const mediaQueryList = window.matchMedia(PREFERSREDUCEDMOTION);
    setPrefersReducedMotion(!mediaQueryList.matches);
    const listener = (event) => {
      setPrefersReducedMotion(!event.matches);
    };
    mediaQueryList.addEventListener("change", listener);
    return () => {
      mediaQueryList.removeEventListener("change", listener);
    };
  }, []);
  return prefersReducedMotion;
}
function useComponentInactiveLogging(componentName, disabled, isMotionReduced, isLoggingEnabled) {
  useEffect(() => {
    if (isLoggingEnabled && disabled)
      logWarning(`${componentName} is disabled as disabled='true'`);
  }, [disabled, isLoggingEnabled, componentName]);
  useEffect(() => {
    if (isLoggingEnabled && isMotionReduced)
      logWarning(
        `${componentName} is disabled as prefers-reduced-motion is set to 'reduce'`
      );
  }, [isMotionReduced, isLoggingEnabled, componentName]);
}

// utils/magic-animation.ts
var import_react_is = __toESM(require_react_is());
import { Children, createElement, isValidElement } from "react";
import { isMotionComponent, m } from "framer-motion";
function isNodeText(node) {
  return typeof node === "string" || Array.isArray(node) && node.every((child) => typeof child === "string");
}
function getIfComponentCanBeConverted(node, nodeType) {
  return !isMotionComponent(nodeType) && !(0, import_react_is.isPortal)(node) && !(0, import_react_is.isProfiler)(node) && !(0, import_react_is.isForwardRef)(node) && !(0, import_react_is.isContextProvider)(node) && !(0, import_react_is.isContextConsumer)(node);
}
function getLayoutValueFromChildren(children) {
  if (isNodeText(children)) {
    return "position";
  }
  return true;
}
var forbiddenComponentKeys = /* @__PURE__ */ new Set([
  "MagicExit",
  "MagicMotion",
  "MagicTabSelect"
]);
function handleForbiddenComponent(node) {
  if (typeof node.type === "function") {
    if (node.key !== null && forbiddenComponentKeys.has(String(node.key))) {
      return node;
    }
    return null;
  }
  return null;
}
function convertChildrenToMotionChildren(children, customProps, nodeInfo) {
  const { isRootNode, rootNodeCallback, isLoggingEnabled, depth } = nodeInfo;
  if (isLoggingEnabled && children) {
    const numOfDashes = (depth - 1) * 2;
    console.groupCollapsed(
      `${" ".repeat(numOfDashes)}\u{1F449} Depth: ${depth},`,
      "elements:",
      children
    );
  }
  const animatedChildren = Children.map(children, (child) => {
    let node = child;
    if (isLoggingEnabled)
      console.log(node);
    if (!isValidElement(node) || node.key === "exclude") {
      if (isLoggingEnabled && isValidElement(node) && node.key === "exclude")
        logWarning(KEYEXCLUDEMESSAGE);
      return node;
    }
    let parent = null;
    while (typeof node.type === "function") {
      if (isLoggingEnabled)
        logSuccess(FUNCTIONCOMPONENTMESSAGE(node.type.name));
      const nodeProps = node.props;
      parent = node;
      node = node.type(nodeProps);
      if (node) {
        const forbiddenResult = handleForbiddenComponent(node);
        if (forbiddenResult) {
          if (isLoggingEnabled)
            logWarning(FORBIDDENELEMENTMESSAGE(node.key));
          return parent;
        }
      }
      if (!isValidElement(node))
        return node;
    }
    const childType = node.type;
    const nodeRef = (0, import_react_is.isPortal)(node) ? null : node.ref;
    const typeOfNewElement = getIfComponentCanBeConverted(node, node.type) ? m[childType] : node.type;
    const newElemChildren = convertChildrenToMotionChildren(
      node.props.children,
      customProps,
      { isRootNode: false, isLoggingEnabled, depth: depth + 1 }
    );
    const newElem = createElement(
      typeOfNewElement,
      __spreadProps(__spreadValues(__spreadProps(__spreadValues({}, node.props), {
        ref: nodeRef
      }), customProps), {
        layout: getLayoutValueFromChildren(node.props.children)
      }),
      newElemChildren
    );
    if (isRootNode && rootNodeCallback) {
      return rootNodeCallback(newElem);
    }
    return newElem;
  });
  if (isLoggingEnabled && children) {
    console.groupEnd();
  }
  return animatedChildren;
}

// magic-card/index.tsx
import { jsx, jsxs } from "react/jsx-runtime";
function MagicCard({
  children,
  isCardExpanded,
  transition,
  isBackgroundFadeEnabled = true,
  onBackgroundFadeClick
}) {
  const rootNode = useRef(null);
  function convertChildrenToMagicCardChildren(nodes, depth) {
    return Children2.map(nodes, (child, i) => {
      let node = child;
      if (!isValidElement2(node))
        return node;
      let isRoot = false;
      if (depth === 1 && i === 0) {
        isRoot = true;
      }
      if (typeof node.type === "function") {
        node = node.type(node.props);
        if (!isValidElement2(node))
          return node;
      }
      const nodeType = node.type;
      const nodeProps = node.props;
      const _a = nodeProps, { className } = _a, restOfProps = __objRest(_a, ["className"]);
      const newElem = createElement2(
        m2[nodeType],
        __spreadProps(__spreadValues({}, restOfProps), {
          ref: isRoot ? rootNode : void 0,
          className: `${className != null ? className : ""} ${isRoot && isCardExpanded ? "react-magic-motion-card-expanded" : "react-magic-motion-card-condensed"}`,
          layout: getLayoutValueFromChildren(nodeProps.children),
          transition: isRoot ? transition : void 0
        }),
        convertChildrenToMagicCardChildren(nodeProps.children, depth + 1)
      );
      return newElem;
    });
  }
  const motionChildren = convertChildrenToMagicCardChildren(children, 1);
  const { placeholderBoxHeight, placeholderBoxWidth } = usePlaceholderBoxSize(
    isCardExpanded,
    rootNode
  );
  return /* @__PURE__ */ jsxs(LazyMotion, { features: domMax, children: [
    motionChildren,
    isCardExpanded && /* @__PURE__ */ jsx(
      "div",
      {
        className: "react-magic-motion-placeholder-box",
        style: {
          width: placeholderBoxWidth,
          height: placeholderBoxHeight
        }
      }
    ),
    isBackgroundFadeEnabled && /* @__PURE__ */ jsx(AnimatePresence, { children: isCardExpanded && /* @__PURE__ */ jsx(
      m2.div,
      {
        initial: { opacity: 0 },
        animate: { opacity: 1 },
        exit: { opacity: 0 },
        onClick: onBackgroundFadeClick,
        className: "react-magic-motion-background-fade"
      }
    ) })
  ] });
}

// magic-motion/index.tsx
import {
  LazyMotion as LazyMotion2,
  domMax as domMax2,
  LayoutGroup
} from "framer-motion";
import { jsx as jsx2 } from "react/jsx-runtime";
function MagicMotion({
  children,
  transition,
  layoutDependency,
  disabled,
  isLoggingEnabled
}) {
  const isMotionReduced = usePrefersReducedMotion();
  const motionChildren = convertChildrenToMotionChildren(
    children,
    {
      layoutDependency,
      transition
    },
    { isRootNode: true, isLoggingEnabled, depth: 1 }
  );
  useComponentInactiveLogging(
    "MagicMotion",
    disabled,
    isMotionReduced,
    isLoggingEnabled
  );
  return isMotionReduced || disabled ? children : /* @__PURE__ */ jsx2(LayoutGroup, { children: /* @__PURE__ */ jsx2(LazyMotion2, { features: domMax2, children: motionChildren }) }, "MagicMotion");
}

// magic-tab-select/index.tsx
import { m as m3, LazyMotion as LazyMotion3, domMax as domMax3 } from "framer-motion";
import { createElement as createElement3 } from "react";
import { Fragment, jsx as jsx3 } from "react/jsx-runtime";
function MagicTabSelect({
  children,
  id,
  transition,
  disabled = false
}) {
  let motionChildren = children;
  const isMotionReduced = usePrefersReducedMotion();
  if (typeof motionChildren.type === "function") {
    motionChildren = motionChildren.type(
      motionChildren.props
    );
    if (motionChildren === null)
      motionChildren = /* @__PURE__ */ jsx3(Fragment, {});
  }
  const childType = motionChildren.type;
  const motionElement = createElement3(
    m3[childType],
    __spreadProps(__spreadValues({}, motionChildren.props), { layoutId: id, transition }),
    motionChildren.props.children
  );
  return isMotionReduced || disabled ? children : /* @__PURE__ */ jsx3(LazyMotion3, { features: domMax3, children: motionElement }, "MagicTabSelect");
}

// magic-exit/index.tsx
import { AnimatePresence as AnimatePresence2 } from "framer-motion";
import { cloneElement } from "react";
import { Fragment as Fragment2, jsx as jsx4 } from "react/jsx-runtime";
function MagicExit({
  children,
  initial,
  animate,
  exit,
  mode = "sync",
  disabled,
  isLoggingEnabled
}) {
  const isMotionReduced = usePrefersReducedMotion();
  const motionChildren = convertChildrenToMotionChildren(
    children,
    {},
    {
      isRootNode: true,
      rootNodeCallback: (rootElem) => {
        const clonedRootElem = cloneElement(rootElem, __spreadProps(__spreadValues({}, rootElem.props), {
          initial,
          animate,
          exit
        }));
        return clonedRootElem;
      },
      isLoggingEnabled,
      depth: 1
    }
  );
  useComponentInactiveLogging(
    "MagicExit",
    disabled,
    isMotionReduced,
    isLoggingEnabled
  );
  return disabled || isMotionReduced ? /* @__PURE__ */ jsx4(Fragment2, { children }) : /* @__PURE__ */ jsx4(AnimatePresence2, { mode, children: motionChildren }, "MagicExit");
}

// index.tsx
export * from "framer-motion";
export {
  MagicCard,
  MagicExit,
  MagicMotion,
  MagicTabSelect
};
/*! Bundled license information:

react-is/cjs/react-is.production.min.js:
  (**
   * @license React
   * react-is.production.min.js
   *
   * Copyright (c) Facebook, Inc. and its affiliates.
   *
   * This source code is licensed under the MIT license found in the
   * LICENSE file in the root directory of this source tree.
   *)

react-is/cjs/react-is.development.js:
  (**
   * @license React
   * react-is.development.js
   *
   * Copyright (c) Facebook, Inc. and its affiliates.
   *
   * This source code is licensed under the MIT license found in the
   * LICENSE file in the root directory of this source tree.
   *)
*/
